app/models/course/assessment/question/programming.rb

```diff
@@ -44,6 +44,25 @@ class Course::Assessment::Question::Programming < ApplicationRecord # rubocop:di
   has_many :test_cases, class_name: 'Course::Assessment::Question::ProgrammingTestCase',
                         dependent: :destroy, foreign_key: :question_id, inverse_of: :question
 
```

Comment:
Actually why not just add a new `validates` after line 34, instead of making an implicit validation in `assign_attributes`?  
This way, you won't have to change `update` in `proramming_controller.rb`

Reply:
Changed to validate with a function under line 34